src/cart_cell.h

```diff
@@ -0,0 +1,269 @@
+/*
+ * Copyright 2025 compiler-research.org, Salvador de la Torre Gonzalez, Luciana
+ * Melina Luque
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *     SPDX-License-Identifier: Apache-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * This file contains a model developed under Google Summer of Code (GSoC)
+ * for the compiler-research.org organization.
+ */
+
+#ifndef CART_CELL_H_
+#define CART_CELL_H_
+
+#include "tumor_cell.h"
+#include "core/agent/agent.h"
+#include "core/agent/cell.h"
+#include "core/agent/new_agent_event.h"
+#include "core/behavior/behavior.h"
+#include "core/container/math_array.h"
+#include "core/diffusion/diffusion_grid.h"
+#include "core/interaction_force.h"
+#include "core/real_t.h"
+
+namespace bdm {
+
+/// Enumeration defining the possible states of a CAR-T cell
+///
+/// This enum class represents the different states that a CAR-T cell can be in
+/// during its lifecycle in the simulation.
+enum class CartCellState : int {
+  kAlive =
+      0,  ///< Living cell state - the cell is alive and functioning normally
+
+  kApoptotic =
+      1  ///< Apoptotic phase - the cell is undergoing programmed cell death
+         ///< characterized by cell shrinkage and controlled death
+};
+
+/// CAR-T cell class implementation
+///
+/// This class represents a CAR-T (Chimeric Antigen Receptor T-cell) in the
+/// simulation. It inherits from the base Cell class and includes specific
+/// behaviors and properties related to CAR-T cell biology, including states,
+/// volume dynamics, and interactions with tumor cells.
+class CartCell : public Cell {
```

Comment:
I'd think a better name would be `CarTCell` so that it doesn't read as a shop cart.